eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/OnlyFinalFieldsForImmutableClassCheck.java

```diff
@@ -0,0 +1,97 @@
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.AnnotationUtil;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import java.util.Optional;
+
+/**
+ * Verify the classes with annotation {@code @Immutable} should have following rules:
+ * <ol>
+ *   <li>Only final fields allowed</li>
+ * </ol>
+ */
+public class OnlyFinalFieldsForImmutableClassCheck extends AbstractCheck {
+    private static final String IMMUTABLE_NOTATION = "Immutable";
+    private static final String ERROR_MSG = "The variable field ''%s'' should be final." +
```

Comment:
I thought our checkstyle rules had the + sign on the next line for strings. or maybe I'm misremembering... 🤔